core/src/main/scala/filodb.core/query/RangeVector.scala

```diff
@@ -179,7 +181,7 @@ object SerializableRangeVector extends StrictLogging {
     try {
       OffheapLFSortedIDMap.validateNoSharedLocks()
       val rows = rv.rows
-      while (rows.hasNext) {
+      while (numRows < limit && rows.hasNext) {
```

Comment:
I think it's better to apply the limit not here in the SRV but in the MetadataExecPlan stuff, and I think it should be similar to the limit that Vish has in place.  In fact, why doesn't Vish's limits help here?

Reply:
@velvia  

1. timeseries queries need to have higher limits and metadata queries have lower limits.
2. Next, metadata queries should not throw an exception when limit is reached, whereas for timeseries data we should not need to throw exception.
3. Metadata queries just return one Range Vector, so limit has to be applied when constructing SRV.
3. The existing limit implementation for timeseries data can be slightly improved by taking remaining samples into the SRV factory and limiting the iteration there. The limit may happen sooner that current implementation.